src/viam/logging.py

```diff
@@ -1,11 +1,43 @@
+import asyncio
 import logging
 import sys
 from copy import copy
-from logging import DEBUG, ERROR, FATAL, INFO, WARN, WARNING  # noqa: F401
+from datetime import datetime
+from logging import DEBUG, ERROR, FATAL, INFO, WARN, WARNING, LogRecord  # noqa: F401
 from typing import Dict
 
+import viam
+
 LOG_LEVEL = INFO
 LOGGERS: Dict[str, logging.Logger] = {}
+_MODULE_PARENT = None
+
+
+class ModuleHandler(logging.Handler):
+    def __init__(self, logger: logging.Logger):
+        self.logger = logger
+        self.parent = _MODULE_PARENT
+
+        logger.handlers.clear()
+        super().__init__()
+
+    def emit(self, record: LogRecord):
+        assert isinstance(record, logging.LogRecord)
+        stack = f"exc_info: {record.exc_info}, exc_text: {record.exc_text}, stack_info: {record.stack_info}"
+        message = f"{record.filename}:{record.lineno}\t{record.msg}"
+        time = datetime.fromtimestamp(record.created)
+
+        try:
+            assert self.parent is not None
+            asyncio.create_task(
+                self.parent.log(record.name, record.levelname, time, message, stack), name=f"{viam._TASK_PREFIX}-LOG-{time}"
+            )
+        except Exception as err:
+            self.logger.handlers.clear()
+            addHandlers(self.logger, True)
```

Comment:
Why are we clearing and re-adding handlers in the event of a failure to log?

Reply:
I realized that the clear is redundant since `addHandlers` clears them for you. But if module logging errors, then we want the logger to have the regular handlers so that it can log using stderr and stdout. I wanted to readd the `ModuleHandler` after that so that the logger doesn't just never try module logging again.

Reply:
Ah gotcha that makes sense; thanks for context! Can we leave a comment here explaining that and remove the redundant `clear()` call?